src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -52,13 +52,20 @@ export const PartyMap: React.FC = () => {
         selectRoom={selectRoom}
         unselectRoom={unselectRoom}
       />
-
-      <RoomModal
-        room={selectedRoom}
-        venue={currentVenue}
-        show={hasSelectedRoom}
-        onHide={unselectRoom}
-      />
+      {selectedRoom?.type === RoomTypes.video ? (
+        <RoomModalVideo
+          room={selectedRoom}
+          show={hasSelectedRoom}
+          onHide={unselectRoom}
+        />
+      ) : (
+        <RoomModal
+          room={selectedRoom}
+          venue={currentVenue}
+          show={hasSelectedRoom}
+          onHide={unselectRoom}
+        />
+      )}
```

Comment:
[note] Here you render either `RoomModalVideo` or `RoomModal` based on the `RoomTypes.video`

Reply:
Instead of handling this different room type selection at the `PartyMap` layer, I would expect it's more of a distinction to be handled within the `RoomModal` component itself. That's where we already handle `RoomTypes.unclickable` currently.

Reply:
```suggestion
      <RoomModal
        room={selectedRoom}
        venue={currentVenue}
        show={hasSelectedRoom}
        onHide={unselectRoom}
      />
```

Reply:
So that we can reuse `IntroVideoPreviewModal` in its entirety, within `RoomModal` I would add a `switch`/`case` (similar to what is done in `src/pages/VenuePage/TemplateWrapper.tsx`) that switches on `room.type`, and chooses to render either `IntroVideoPreviewModal` (if it's `RoomTypes.video`), or `RoomModalContent` (as the `default` case)

This would end up looking roughly something like this:

```typescript
switch (room.type) {
  case RoomTypes.video:
    return <IntroVideoPreviewModal introVideoUrl={room?.url + `?autoplay=1`} isVisible={show} onHide={onHide}>

  default:
    return <RoomModalContent room={room} venueName={venue.name}>
}
```

Reply:
Even better than using

```
room?.url + `?autoplay=1`
```

might be to use our `ConvertToEmbeddableUrl` util function that already handles this sort of thing?

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/ConvertToEmbeddableUrl.tsx

Reply:
Searching for usages of `ConvertToEmbeddableUrl`:

- https://github.com/sparkletown/sparkle/search?q=ConvertToEmbeddableUrl

I came across `VideoModal`, which might be an even better choice than `IntroVideoPreviewModal` here:

- https://github.com/sparkletown/sparkle/blob/staging/src/components/organisms/VideoModal/VideoModal.tsx

---

@mike-lvov What do you think? Any reason you would/wouldn't use `VideoModal` instead of `IntroVideoPreviewModal` here?

Reply:
Implemented in new completely refactored version